.buildkite/dagster-buildkite/dagster_buildkite/repository.py

```diff
@@ -0,0 +1,23 @@
+import logging
+from pathlib import Path
+from typing import Set
+
+from dagster_buildkite.git import ChangedFiles
+from dagster_buildkite.python_packages import PythonPackages
+
+
+class Repositories:
+    all: Set[Path] = set()
+
+    @classmethod
+    def load_repository(cls, git_repository_directory: Path = Path(".")) -> None:
+        # Memoize
+        if git_repository_directory in cls.all:
+            return None
+
+        logging.info(f"Parsing repo {git_repository_directory.absolute()}")
+
+        ChangedFiles.load_repository(git_repository_directory)
+        PythonPackages.load_repository(git_repository_directory)
+
+        cls.all.add(git_repository_directory)
```

Comment:
this worth it? feels superfluous

Reply:
There's some ordering that's important here - PythonPackages depends on ChangedFiles. I guess I could move the `ChangedFiles.load_repository()` into PythonPackages though.

Reply:
But it does give us a nice place to group related actions for logging purposes.

Reply:
maybe just collapse everything in to one class?

Reply:
I ended up making the dependency explicit within PythonPackages.